src/Symfony/Bundle/FrameworkBundle/Command/CachePoolClearCommand.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\HttpKernel\CacheClearer\Psr6CacheClearer;
+
+/**
+ * Clear cache pools.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class CachePoolClearCommand extends ContainerAwareCommand
```

Comment:
> I like the current way as we have a very clear BC promise (and we can probably add some specific notes about build-in commands, controllers, ...)

This is not clear until you specify the list of classes which are risky to inherit. What does "want to go wild" mean? People should know what "wild" is. For some inheritance-oriented people extending listener/command is OK, they think it is very OOPish and, therefore, valid. 

I think `final` is easier way to make BC promise truly clear.

> I'm not sold yet on adding final as it's a hard stop on inheritance and extensibility

Yes, because you want to stop it on purpose, don't you? On other hand, you want to see private members by default, it leads to classes with no protected members at all. This policy is already "hard stop" on inheritance and extensibility: it makes those classes pointless to inherit.

Inheritance is not the only way to extend behaviour of the class. In most cases, you can fix your problem with composition.

Also, keeping classes open to inheritance blocks PRs like this one: #11708.